### ISSUE TITLE
ReLU on Discriminator activations?

### ISSUE THREAD

**Issue by neel-dey** [google/lecam-gan#1]
Thank you for making your implementation available, I'm quite excited to apply the regularizer to ongoing projects.

There was some [discussion online](https://twitter.com/neeldey/status/1379462008215646215) recently about a possible mismatch between Eqn 4 in a public preprint of the paper ([available here](https://faculty.ucmerced.edu/mhyang/papers/cvpr2021_gan_limited_data.pdf)) and the code implementation at these lines: {[tf](https://t.co/6vcZjKfdWx?amp=1), [pytorch](https://t.co/9Vm6qsJBu3?amp=1)}.

Could you please clarify if there's supposed to be a ReLU (present in the code but not the preprint) in Eqn. 4 as well? Thanks again!

**Comment by hytseng0509** [comment]
Thanks for the interest in this work.

We add the ReLU function in practice to avoid the instability issue in the early training stage.

Take [Pytorch](https://github.com/google/lecam-gan/blob/f268aa68eb99e8e385ad26c18cafc276afae922c/biggan_cifar/losses.py#L20) for example (also Figure 8 in the paper), in the very early training stage, the discriminator predictions of fake images may be larger than those of real images (i.e. `dis_real` < `ema.D_fake` and `ema.D_real` < `dis_fake`). Since this is not the case we consider (we only consider the normal case, i.e., discriminator predictions of fake images smaller than those of real ones), we use the ReLU trick to avoid backpropagating our regularization loss in this rare scenario.

**Comment by neel-dey** [comment]
Makes sense, thanks for the clarification.
